applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -158,146 +158,180 @@ void File::GetAttributeNames(std::string ObjectPath, std::vector<std::string>& r
-            KRATOS_ERROR_IF_NOT(IsGroup(sub_path))
-                << "Path exists and is not a group: " << sub_path << std::endl;
+        // Get size of name.
+        ssize_t ssize;
+        ssize = H5Lget_name_by_idx(m_file_id, rGroupPath.c_str(), H5_INDEX_NAME,
+                                    H5_ITER_INC, i, buffer, max_ssize, H5P_DEFAULT);
+        KRATOS_ERROR_IF(ssize < 0) << "H5Lget_name_by_idx failed." << std::endl;
+        KRATOS_ERROR_IF(ssize > max_ssize) << "Link name size exceeds "
+                                           << max_ssize << std::endl;
+        rNames[i].resize(ssize);
+        std::copy_n(buffer, ssize, rNames[i].begin());
     }
-}
-
-void File::WriteDataSet(std::string Path, const Vector<int>& rData)
-{
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
+    KRATOS_ERROR_IF(H5Gclose(group_id) < 0) << "H5Gclose failed." << std::endl;
     KRATOS_CATCH("");
 }
 
-void File::WriteDataSet(std::string Path, const Vector<double>& rData)
+void File::GetGroupNames(const std::string& rGroupPath, std::vector<std::string>& rNames) const
 {
     KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
-}
 
-void File::WriteDataSet(std::string Path, const Vector<array_1d<double, 3>>& rData)
-{
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
+    rNames.resize(0);
+    std::vector<std::string> link_names;
+    GetLinkNames(rGroupPath, link_names);
+    rNames.reserve(link_names.size());
+    for (const auto& r_name : link_names)
+        if (IsGroup(rGroupPath + '/' + r_name))
+            rNames.push_back(r_name);
+
     KRATOS_CATCH("");
 }
 
-void File::WriteDataSet(std::string Path, const Matrix<int>& rData)
+void File::AddPath(const std::string& rPath)
 {
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
+    KRATOS_ERROR_IF(Internals::IsPath(rPath) == false) << "Invalid path: " <<rPath << std::endl;
+
+    std::vector<std::string> splitted_path = Internals::Split(rPath, '/');
+    std::string sub_path;
+    for (const auto& r_link: splitted_path)
+    {
+        sub_path += '/' + r_link;
+        if (HasPath(sub_path) == false)
+            CreateGroup(sub_path); // Add missing link.
+        else
+            KRATOS_ERROR_IF_NOT(IsGroup(sub_path))
+                << "Path exists and is not a group: " << sub_path << std::endl;
+    }
 }
 
-void File::WriteDataSet(std::string Path, const Matrix<double>& rData)
+void File::WriteAttribute(const std::string& rObjectPath, const std::string& rName, const std::string& rValue)
 {
     KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
+    boost::timer timer;
```

Comment:
I think it has somethings that can be used to construct a timer mechanism, but this is something we should define in one place (in core). Until we do this, I would leave this as it is. There is a Timer in the core timer.h, but it's not so suitable for timing a single function imo because it uses global variables, which I would prefer to avoid.

Reply:
@msandre I also would avoid using boost::timer, since i think it does fake outputs if you have openmp active. 

c++11 has std::chrono if you need a similar feature, however i am not sure of how it behaves when openmp is active.

The OpenMP timer could be a good option too.

Having said this i definitely do not consider this a blocker, so i am just commenting